feat: create update schematic for cms component state

znikola opened the pull request:
closes #6027

psmul commented on projects/schematics/src/migrations/2_0/validate.ts:
```diff
@@ -0,0 +1,29 @@
+  };
+}
+
+export function validateAngularVersion(angularVersion: string): void {
```
Maybe we can go with something more generic and move it to shared/utils?

znikola replied:
Not sure if this should be a generic method. We could always do that of course, but I don't see it being use in multiple places for now, besides this validation. Do you have some specific use case on mind?

psmul replied:
Tried to come up with some example, but maybe I overreacted with this one :) Let's leave it as is.

znikola replied:
It's a valid proposition :)

znikola commented:
>     1. In file **update-cms-component-state_spec**, line 436 we have unit test with description 'ultimate test'. Can we make it more descriptive?

Sure, I renamed it to _all test cases in one file_.

>     2. When running **yarn build** in schematics source project we get an error about missing type for shelljs. Had to run command `npm install @types/shelljs`.

Good catch, I added it to the schematics' `package.json` and generated a new `yarn.lock`

claassistantio commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/SAP/cloud-commerce-spartacus-storefront?pullRequest=6338) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/SAP/cloud-commerce-spartacus-storefront?pullRequest=6338) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/SAP/cloud-commerce-spartacus-storefront?pullRequest=6338) it.</sub>

dunqan commented on package.json:
```diff
@@ -158,6 +159,7 @@
     "sass": "^1.22.10",
     "semver": "^6.3.0",
     "semver-extra": "^3.0.0",
+    "shelljs": "^0.8.3",
```
Can you help me understand why we need this dependency?

znikola replied:
It's required for spec files, where we use it to switch directories.

dunqan commented on projects/cds/package.json:
```diff
@@ -1,6 +1,6 @@
 {
   "name": "@spartacus/cds",
-  "version": "1.0.0-next.0",
+  "version": "2.0.0",
```
I believe we should set `2.0.0-next.0` for all versions, will it break the logic?
(we should be able to run `ng update @spartacus/schematics@next` until 2.0 becomes the final)

znikola replied:
Done

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/signed)](https://cla-assistant.io/SAP/cloud-commerce-spartacus-storefront?pullRequest=6338) <br/>All committers have signed the CLA.
